src/main/java/atreides/ui/Parser.java

```diff
@@ -0,0 +1,51 @@
+package atreides.ui;
+
+import atreides.command.AddCommand;
+import atreides.command.Command;
+import atreides.command.DeleteCommand;
+import atreides.command.ErrorCommand;
+import atreides.command.ExitCommand;
+import atreides.command.ListCommand;
+import atreides.command.MarkCommand;
+import atreides.command.UnMarkCommand;
+
+import java.util.Set;
+
+/**
+ * Represents the class to parse commands and creates relevant commands
+ */
+public class Parser {
+
+    /**
+     * Parses the message and creates relevant commands
+     * @param msg
+     * @return
+     */
+    public static Command parse(String msg) {
+        Set<String> commands = Set.of("mark", "unmark", "delete", "todo", "event", "deadline", "find");
```

Comment:
I like how for the commands and the words variable later you used descriptive names, it makes it clear what they are for.